src/transformers/generation/candidate_generator.py

```diff
@@ -175,6 +182,7 @@ def get_candidates(self, input_ids: torch.LongTensor) -> Tuple[torch.LongTensor,
         # Don't generate more than `max_length - 1` candidates since the target model generates one extra token.
         new_cur_len = input_ids.shape[-1]
         max_new_tokens = min(int(self.num_assistant_tokens), self.generation_config.max_length - new_cur_len - 1)
+        min_new_tokens = min(max_new_tokens, self.min_length - new_cur_len)
```

Comment:
Won't this result in negative values? 

If I've generated more tokens than `self.min_length` i.e. `new_cur_len > self.min_length`

Reply:
good point, I've missed it